From 8b266132ee180ff8cc17b394ba64850f207a3dab Mon Sep 17 00:00:00 2001
From: Harmen Stoppels <harmenstoppels@gmail.com>
Date: Tue, 12 Jan 2021 12:26:44 +0100
Subject: [PATCH] Drop clang rt builtins linking on hip::host

Ref this comment
https://github.com/ROCm-Developer-Tools/HIP/pull/2217#discussion_r555402581:

> Yea, this should only be linked on hip::device as I believe its needed
> for __fp16 support on the GPU. If users need this on the host-side,
> then they should explicitly add it.
---
 hip-config.cmake.in | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/hip-config.cmake.in b/hip-config.cmake.in
index 6dfbd527..0ac1b659 100644
--- a/hip-config.cmake.in
+++ b/hip-config.cmake.in
@@ -206,9 +206,6 @@ if(HIP_COMPILER STREQUAL "clang")
   endif()
 
   # Add support for __fp16 and _Float16, explicitly link with compiler-rt
-  set_property(TARGET hip::host APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
-  )
   set_property(TARGET hip::device APPEND PROPERTY
     INTERFACE_LINK_LIBRARIES -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
   )
-- 
2.25.1

